docs/structure.rst

```diff
@@ -0,0 +1,43 @@
+
+Structure of a Reactive Charm
+-----------------------------
+
+The structure of a reactive charm is similar to existing charms, with the
+addition of ``reactive`` directory and the ``relations`` directory under
+``hooks``:
+
+.. code-block:: text
+
+    .
+    ├── metadata.yaml
+    ├── reactive
+    │   └── common.py
+    └── hooks
+        ├── pgsql-relation-changed
+        └── relations
+            └── pgsql
+                ├── interface.yaml
+                ├── peer.py
+                ├── provides.py
+                └── requires.py
+
+The hooks will need to call :func:`reactive.main() <charms.reactive.main>`,
+and the decorated handler blocks can be placed in any file under the ``reactive``
+directory.  Thus, pretty much all of your hooks will end up contain little more
+than:
```

Comment:
This is talking as if charm authors create the hooks, whereas the reality is they are pulled in from the base layer. Its nice to explain the mechanism, but it needs to be clearer to charm authors that they are created automatically at charm build time.